test/ChainRulesTests.jl

```diff
@@ -0,0 +1,75 @@
+module ChainRulesTest
+
+using LinearAlgebra
+using ChainRules
+using DiffResults
+using ReverseDiff
+using Test
+
+f(x) = sum(4x .+ 1)
+
+function ChainRules.rrule(::typeof(f), x)
+    r = f(x)
+    function back(d)
+        #=
+        The proper derivative of `f` is 4, but in order to
+        check if `ChainRules.rrule` had taken over the compuation,
+        we define a rrule that returns 3 as `f`'s derivative.
+
+        After importing this rrule into ReverseDiff, if we get 3
+        rather than 4 when we compute the derivative of `f`, it means
+        the importing mechanism works.
+        =#
+        return ChainRules.NoTangent(), fill(3 * d, size(x))
+    end
+    return r, back
+end
+
+ReverseDiff.@grad_from_chainrules f(::ReverseDiff.TrackedArray)
+
+begin # test ChainRules.rrule
```

Comment:
`begin` could/should be removed and possibly be replaced with `@testset`. It does not introduce a new scope, if this was the intention.

Reply:
updated.